src/pickup.c

```diff
@@ -3207,258 +3208,344 @@ static const char tippables[] = { ALL_CLASSES, TOOL_CLASS, 0 };
+		tipcontainer(cobj);
+		return 1;
+	}
+	/* assorted other cases */
+	if (Is_candle(cobj) && cobj->lamplit) {
+		/* note "wax" even for tallow candles to avoid giving away info */
+		spillage = "wax";
+	} else if ((cobj->otyp == POT_OIL && cobj->lamplit)
+			   || (cobj->otyp == OIL_LAMP && cobj->age != 0L)
+			   || (cobj->otyp == MAGIC_LAMP && cobj->spe != 0)) {
+		spillage = "oil";
+		/* todo: reduce potion's remaining burn timer or oil lamp's fuel */
+	} else if (cobj->otyp == CAN_OF_GREASE && cobj->spe > 0) {
+		/* charged consumed below */
+		spillage = "grease";
+	} else if (cobj->otyp == FOOD_RATION || cobj->otyp == CRAM_RATION
+			   || cobj->otyp == LEMBAS_WAFER) {
+		spillage = "crumbs";
+	} else if (cobj->oclass == VENOM_CLASS) {
+		spillage = "venom";
+	}
+	if (spillage) {
+		buf[0] = '\0';
+		if (is_pool(u.ux, u.uy, TRUE))
+			Sprintf(buf, " and gradually %s", vtense(spillage, "dissipate"));
+		else if (is_lava(u.ux, u.uy))
+			Sprintf(buf, " and immediately %s away",
+					vtense(spillage, "burn"));
+		pline("Some %s %s onto the %s%s.", spillage,
+			  vtense(spillage, "spill"), surface(u.ux, u.uy), buf);
+		/* shop usage message comes after the spill message */
+		if (cobj->otyp == CAN_OF_GREASE && cobj->spe > 0) {
+			consume_obj_charge(cobj, TRUE);
+		}
+		/* something [useless] happened */
+		return 1;
+	}
+	/* anything not covered yet */
+	if (cobj->oclass == POTION_CLASS) /* can't pour potions... */
+		pline_The("%s %s securely sealed.", xname(cobj), otense(cobj, "are"));
+	else if (uarmh && cobj == uarmh)
+		return tiphat();
+	else if (cobj->otyp == STATUE)
+		pline("Nothing interesting happens.");
+	else
+		pline1(nothing_happens);
+tipmonster:
+#define can_respond(mtmp) (mtmp && mtmp->mcanmove && !mtmp->msleeping && m_canseeu(mtmp))
+	if (getdir("Tip a monster? (in what direction)")){
+		struct monst* mtmp = m_at(u.ux+u.dx, u.uy+u.dy);
+		if (!canspotmon(mtmp))
+			mtmp = (struct monst *) 0;
+		
+		if (!mtmp)
+			You("don't see anybody there to tip.");
+		else if (mtmp->mtyp == PM_ROTHE){
+			if (near_capacity() >= MOD_ENCUMBER)
+				You("are carrying too much to do that without falling over.");
+			else if (!(mtmp->mfrozen || mtmp->msleeping))
+				pline("The rothe is awake and resists being toppled over!");
+			else {
+				You("try to push the rothe over.");
+				// going to assume that even if you have no hands, you can still bodycheck
+				// the rothe into wakefulness
+				if (ACURRSTR < 14){
+					pline("It's too heavy for you to budge.");
+					if (rn2(2)){
+						mtmp->msleeping = 0;
+						mtmp->mfrozen = 0;
+						pline("At your push, the rothe wakes up!");
+					}
+				} else {
+					if (Hallucination)
+						pline("The creature has fallen and can't get up.");
+					else
+						You("successfully tip the rothe onto its side, startling it awake!");
+					
+					mtmp->msleeping = 0;
+					mtmp->mfrozen = 1;
+				}
+			}
+		} else if (mtmp->isshk){
+			if (u.ugold){
```

Comment:
This doesn't seem to have a can_respond(mtmp) check?

Also, dnethack has otherwise maintained compatibility with gold objects for some reason, which is not the case here.  I don't think it would be hard to adjust it.